weatherman.py

```diff
@@ -1,214 +1,362 @@
+    """
+    # MIN HUMIDITY
+
+    month_min_humidity = []
+    min_list = []
+
+    for i in range(len(humidity)):
+        # iterate over list and access the sublist's MAXHumidity element and store it in separate list
+        min_list.append(humidity[i][9])
+        # convert the list into Integer
+    min_list = list(map(int, set(filter(None, min_list))))
+
+    if not min_list:
+        # if there is no MINHumidity value in list then return 1000
+        return 10000
+    else:
+        # find minimum value in list
+        minimum_num = min(min_list )
+        temp_min = str(minimum_num)
+        # minimum temperature we get is of one month.
+        # append the minimum humidity and it's date in month_min_list and return list
+        for item in range(len(humidity)):
+            if humidity[item][9] == temp_min:
+                month_min_humidity.append(humidity[item][0])
+                month_min_humidity.append(temp_min)
+                break
+        return month_min_humidity
+
+
+def year_hottest_days():
+    """
+    it will show the the Hottest days with year and complete date
+
+    """
+    print(str("Year").rjust(4),
+          str("Date").rjust(10),
+          str("Temp").rjust(11),
+          "[Report#2][Hottest Days Report]".rjust(8)
+          )
+    print("-"*80)
+    # iterate over dictionary key @Year e.g 1996 and display it's value.Uses same key for two dictionaries
+    # e.g 1996:1996-12-1 and 1996:49
+    for i in max_temp_dict.keys():
+        print(str(i).rjust(4),
+              str(complete_date_dict[i]).rjust(12),
+              str(max_temp_dict[i]).rjust(8)
+              )
+
+
+def annual_report():
+    """
+    It will show the annual report to the user
+
+    """
+
+    print(str("Year").rjust(7),
+          str("MAXTemp").rjust(7),
+          str("MINTemp").rjust(7),
+          str("MAXHumidity").rjust(7),
+          str("MINHumidity").rjust(7),
+          "[Report#1][Annual Report]".rjust(8)
+          )
+    print("-"*80)
+    # iterate over dictionary key @Year e.g 1996 and display it's value.Uses same key for all dictionaries
+    for i in max_temp_dict.keys():
+        print(str(i).rjust(6),
+              str(max_temp_dict[i]).rjust(6),
+              str(min_temp_dict[i]).rjust(6),
+              str(max_humidity_dict[i]).rjust(8),
+              str(min_humidity_dict[i]).rjust(9)
+              )
+
+
+def main_function(path):
+    """
+    Performing as a Main Function which will call all the other functions
+    :param path: default directory of the Project where all weather files are located
+
+    """
+
+    # Fetch all files in directory
+    files = glob.glob(path)
+    for file in files:
+        # iterate over all files
+        lines = []
+        L = []
+        with open(file) as f:
+            # Skip the first two lines
+            f.__next__()
+            f.__next__()
+            # Read all from a file and store it in List @lines
+            lines = csv.reader(f)

```

Comment:
Yes i will extract using DictReader. Will also make it more memory efficient. Thanks for reviewing.